data-access/UserRepository.cs

```diff
@@ -39,19 +47,45 @@ public UserRepository(IConfiguration config) : base(config)
         /// <param name="userName">The user's username</param>
         /// <param name="password">The user's password</param>
         /// <returns></returns>
-        public async Task<User> GetUserByPasswordAsync(string userName, string password)
+        public async Task<UserLoginResponse> GetUserByPasswordAsync(string userName, string password)
         {
-            var sql = @"SELECT user_name, user_id FROM user WHERE user_name = @UserName AND pw = SHA2(@Password, 224) AND authenticated = 1";
             using (var connection = OpenConnection())
             {
-                var results = await connection.QueryAsync<UserQueryResponse>(sql, new
+                return await connection.QuerySingleAsync<UserLoginResponse>(LoginQuery.GetQuery, new
                 {
                     UserName = userName,
                     Password = password
                 });
+            }
+        }
+
+        public async Task<DetailedUserToken> GetDetailedUserById(UserInfo userInfo)
```

Comment:
OK, now there are too many repository methods, too, and probably controller endpoints. Why don't we have just two user DTOs - one with all the fields and the other with all the fields *and* the token, and we always return all of them? Everything will be simpler - the API documentation, the queries, the repository, the service, the controller.